tests/trx_generator/trx_provider.hpp

```diff
@@ -42,63 +42,64 @@ namespace eosio::testing {
    using fc::time_point;
 
    struct tps_test_stats {
-      uint32_t total_trxs;
-      uint32_t trxs_left;
-
+      uint32_t total_trxs = 0;
+      uint32_t trxs_left = 0;
+      uint32_t trxs_sent = 0;
       time_point start_time;
       time_point expected_end_time;
-   };
+      time_point last_run;
+      time_point next_run;
+      int64_t last_sleep = 0;
 
-   struct simple_tps_monitor {
-      bool monitor_test(const tps_test_stats& stats) {return true;}
    };
 
-   struct null_trx_generator {
-      void generate_and_send() {}
-   };
 
-   constexpr int64_t min_sleep_us = 100;
+   constexpr int64_t min_sleep_us = 1;
 
    template<typename G, typename M>
    struct trx_tps_tester {
```

Comment:
Are you anticipating using `trx_tps_tester` outside of the unit tests?  Should this be how we run and observe generation rate during actual perf runs?  I didn't see it currently incorporated anywhere in the main code path.

Reply:
This will be the piece that calls the refactored main in the trx_generator.